src/Microsoft.DotNet.ProjectModel/ProjectReader.cs

```diff
@@ -246,6 +243,38 @@ public Project ReadProject(Stream stream, string projectName, string projectPath
             return project;
         }
 
+        internal static string NormalizeProjectDirectoryPath(string path)
+        {
+            if (File.Exists(path) &&
+                string.Equals(Path.GetFileName(path), Project.FileName, StringComparison.OrdinalIgnoreCase))
+            {
+                string directoryName = Path.GetDirectoryName(path);
+                if (string.IsNullOrEmpty(directoryName))
+                {
+                    directoryName = Directory.GetCurrentDirectory();
+                }
+
+                return Path.GetFullPath(directoryName);
+            }
+            else if (Directory.Exists(path) &&
+                     File.Exists(Path.Combine(path, Project.FileName)))
+            {
+                return Path.GetFullPath(path);
+            }
+
+            return null;
```

Comment:
> Well.. null can crash the app while string.empty can't. Right?

It's totally up to the caller to crash the app.  Just returning null doesn't crash the app.  The caller could also do `str[0]` which would crash the app.

> I mean, I don't see: that's what we have been doing as a real argument to let's continue doing it.

It is a real argument when you have callers that do this:

```C#
            var normalizedPath = ProjectReader.NormalizeProjectDirectoryPath(projectPath);
            if (normalizedPath == null)
            {
                return null;
            }
```